compiler/crates/relay-compiler/src/build_project/generate_artifacts.rs

```diff
@@ -228,6 +229,30 @@ fn generate_normalization_artifact(
     }
 }
 
+pub fn generate_preloadable_query_parameters_artifact(
+    project_config: &ProjectConfig,
+    normalization: &Arc<OperationDefinition>,
+    id_and_text_hash: &Option<QueryID>,
+    source_keys: Vec<ArtifactSourceKey>,
+    source_file: SourceLocationKey,
+) -> Artifact {
+    let query_id = id_and_text_hash
+        .clone()
+        .expect("Expected operation artifact to have an `id`. Ensure a `persistConfig` is setup for the current project.");
```

Comment:
Theoretically, you don't need query persisting for @prelodable query to work. You can just fetch a query with the text. Yes, it will probably increase the size of that artifact, but still it won't include the normalization part of the query, so it's a code-size win compared to the full normalization artifact.

Reply:
The code generated for `PreloadableQueryRegistry` requires `params.id` at the moment: https://github.com/facebook/relay/blob/54586eccb033cbdd0896bb1d5d53727a523e3032/compiler/crates/relay-compiler/src/artifact_content/content.rs#L347-L350
and `loadQuery` will throw at runtime, if there s no `id`: https://github.com/facebook/relay/blob/54586eccb033cbdd0896bb1d5d53727a523e3032/packages/react-relay/relay-hooks/loadQuery.js#L304-L309

Reply:
Ah, ok... Maybe the right thing to do is to remove these invariants. But for this PR I think it make sense to keep it as is (with only persisted query id). Thanks!